tests/agent/test_lifecycle.py

```
@@ -0,0 +1,224 @@
+                    is_error=False,
+                )
+            )
+        lc.emit(TurnEndEvent(turn_context=TurnContext(1, 1)))
+        lc.emit(AgentEndEvent(status="completed", total_turns=1, total_tools=2))
+        assert lc._depth == 0
+        assert lc.get_totals() == (1, 2)
+
+    def test_wrong_depth_raises(self):
+        """错误嵌套层级抛出 RuntimeError"""
+        lc = Lifecycle()
+        with pytest.raises(RuntimeError, match="AGENT_END.*depth 0.*expected 1"):
```

Ruff (0.15.10)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)